RecoEgamma/EgammaElectronProducers/plugins/GsfElectronCoreBaseProducer.cc

```diff
@@ -13,11 +13,14 @@
 
 using namespace reco;
 
-void GsfElectronCoreBaseProducer::fillDescription(edm::ParameterSetDescription& desc) {
-  desc.add<edm::InputTag>("gsfPfRecTracks", edm::InputTag("pfTrackElec"));
-  desc.add<edm::InputTag>("gsfTracks", edm::InputTag("electronGsfTracks"));
+edm::ParameterSetDescription GsfElectronCoreBaseProducer::makeBaseDescription(std::string const& gsfPfRecTracks,
```

Comment:
why is this copy of a ParameterSetDescription needed?
The old ```fillDescription( ``` was lighter

Reply:
There should be no copy there due to return value optimization. So it's just as light but the input and output is more explicit.

Reply:
Okay! I was not aware that this is a reoccurring pattern, then it should of course stay consistent.